### ISSUE TITLE
Writing Modelica files from an AST

### ISSUE THREAD

**Issue by jbaayen** [pymola/pymola#68]
The ability to write Modelica output files, subsequent to modifying an AST programmatically, would be convenient and help with debugging compilation stages such as flattening.

**Comment by jgoppert** [comment]
I'm curently using modelica xml and the xsd to verify the flat model in the xml branch before I use it for simulation. This sort of has a similar goal, but outputting modelica again would also be useful.

**Comment by jackvreeken** [comment]
If we do add this, it would make sense to mimic OpenModelica's compiler as much as possible. I think I deviated from `omc.exe` with constants a little, but otherwise any ambiguous cases should be handled similarly (at least, that's how I made the flattener). I remember implementing constants a little differently, but I don't really remember how. Over at OpenModelica they also have a huge test bank with input and expected (flattened Modelica) output, which would be a nice way to write tests.

**Comment by jgoppert** [comment]
One reason we might want to use https://github.com/modelica-association/ModelicaXML vs raw modelica for checking the flattened output is that we can use the xsd as a contract and restrict it to exactly what we are willing to process. The goal of modelica xml is to support representations of the compiler process from the original model file all the way to the flattened tree ready for dae simulation, so I think that is pretty aligned with what we want here.

**Comment by jackvreeken** [comment]
Ah, I did not even know that existed. That does look like a good thing to support exporting to (and maybe also importing from?). Is it up to date though (3 years of no activity), because the README says  that their focus has only been on step 4? We don't do scalarized in the CasADi backend, and as far as debugging purposes are concerned, it would be great to spit out an XML of/after step 2.

**Comment by jgoppert** [comment]
@jackvreeken I agree that the importing from would also be valuable. I've already done that here: https://github.com/pymola/pymola/pull/69. I also go over there the 2 options I saw for xml, either ModelicaXML for jmodelica. I ended up leaning towards ModelicaXML because it seemed more flexible and the license. The 3 years of no activity did make me worried as well, but I think it does what we need already, and maybe we can create some more interest if we starting submitting pull requests for modifications/additions as we see fit.
